### DISCUSSION TITLE
Setting language/culture of dialogs

### DISCUSSION THREAD

**Discussion by DineshSolanki** [ookii-dialogs/ookii-dialogs-wpf#50] [Q&A] [answered]
I have tried setting CurrentCulture and CurrentUICulture but none seems to affect the language of dialogs button, could you tell me how to do that.

**Comment by DineshSolanki** [comment, marked as answer]
I have found a way,
we can use SetThreadUILanguage()
for anyone else looking for solution -
```csharp
[DllImport("Kernel32.dll", CharSet = CharSet.Auto)]
static extern System.UInt16 SetThreadUILanguage(System.UInt16 LangId);

SetThreadUILanguage(1036); //french
SetThreadUILanguage((ushort)Thread.CurrentThread.CurrentUICulture.LCID); //to set to currentCulture language
```

you can find language id [here](https://www.autoitscript.com/autoit3/docs/appendix/OSLangCodes.htm)

**Reply by ciarpame** [reply]
Hi, unfortunately this workaround does not seems to be available for projects targeting .NET 4.72
